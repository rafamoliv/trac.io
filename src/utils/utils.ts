/**
 * Util to formate date to locale string
 *
 * @param {Date} DateToFormat
 * @returns '14/12/2022, 21:00:00'
 */

export const dateFormat = (DateToFormat: Date) => {
  if (!DateToFormat) return
  const date = new Date(DateToFormat)
  return date.toLocaleString()
}

/**
 * Util to return initial letters from name and lastname
 *
 * @param {string} name
 * @returns 'RO'
 */

export const initialLettersNameLastname = (name: string) => {
  if (!name) return
  return name
    .match(/(\b\S)?/g)
    .join('')
    .toUpperCase()
}
