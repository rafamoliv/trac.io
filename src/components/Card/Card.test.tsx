import React from 'react'
import i18next from 'i18next'
import { expect, it } from 'vitest'

import { render } from '../../utils/tests'

import { Card } from '.'

const props = {
    children: i18next.t('translation:foo')
}

describe('<Card />', () => {
    describe('Component Snapshot', () => {
        it('render element', () => {
            const { container } = render(<Card {...props} />)
            expect(container.firstChild).toMatchSnapshot()
        })
    })
})
