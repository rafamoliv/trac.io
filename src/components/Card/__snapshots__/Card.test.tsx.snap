// Vitest Snapshot v1

exports[`<Card /> > Component Snapshot > render element 1`] = `
<div
  class="ant-card ant-card-bordered css-dev-only-do-not-override-1me4733"
>
  <div
    class="ant-card-body"
  >
    Welcome to React and react-i18next
  </div>
</div>
`;
